File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
    HashMap<Integer,Integer> tosum=new HashMap<>();
    for(int i=0;i<nums.length;i++){
        int diff=target-nums[i];
        if(tosum.containsKey(diff)&& tosum.get(diff)!=i)
        return new int[]{tosum.get(diff),i};
        tosum.put(nums[i],i);
        }
        return new int[]{};

    }
    }

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        tosum={}
        for i, num in enumerate(nums):
            diff=target-num
            if diff in tosum:
                return [tosum[diff],i]
            tosum[num]=i
        return []